### ISSUE TITLE
sketch plane and grid plane misaligned

### ISSUE THREAD

**Issue by JordanNoone** [KittyCAD/modeling-app#3117]
[Add a title above and insert a description of the issue here]

![Screenshot](https://public.blob.zoo.dev/aa87704c-03db-45d0-8a20-5fb45020d1f7/modeling-app/coredump-20dea63a-e92e-4e5c-9692-1f102ca9523f-screenshot.png)

<details>
<summary><b>Core Dump</b></summary>

[coredump-20dea63a-e92e-4e5c-9692-1f102ca9523f.json](https://public.blob.zoo.dev/9cd8cc54-5462-479c-8197-85f4afee7d95/modeling-app/coredump-20dea63a-e92e-4e5c-9692-1f102ca9523f.json)

Reference ID: 20dea63a-e92e-4e5c-9692-1f102ca9523f
</details>

**Comment by mlfarrell** [comment]
@JordanNoone can you paste in kcl directly for this one?  (easier to reprod vs retyping from an image)

**Comment by mlfarrell** [comment]
likely wont get to this before EOW and possibly siggraph
if someone else on web team can rule out web first (since this is web repo vs engine), that'd be helpful

**Comment by nadr0** [comment]
Able to recreate this
![Image](https://github.com/user-attachments/assets/8860bca2-7b85-44ce-9d41-9eb76e087dbc)

Going to debug the modeling app side to see if it is modeling app or engine issue.

I typed up the source code. This excludes the revolve since there is an engine error when calling the revolve. 
```
const throat_D = 2
const throat_R = throat_D / 2
const throat_flat = 0.1
const diverging_half_angle = 15
const diverging_radius = 0.25
const epsilon = 6
const exit_D = epsilon ^ 0.5 * throat_D
const exit_R = exit_D / 2
const cone_thickness = 0.25
const body_diam = throat_D + 1
const body_R = body_diam / 2
const converging_radius = 0.25
const case_diam = 4
const case_radius = case_diam / 2

const sketch001 = startSketchOn("XZ")
  |> startProfileAt([throat_R, 0], %)
  |> yLine(-throat_flat, %)
  |> arc({
       angleStart: 0,
       angleEnd: diverging_half_angle,
       radius: -diverging_radius
     }, %)
  |> angledLineToX({
       angle: diverging_half_angle - 90,
       to: exit_R
     }, %)
  |> xLine(cone_thickness, %)
  |> angledLineToX({
       angle: diverging_half_angle + 90,
       to: body_R
     }, %)
  |> yLineTo(-.25, %)
  |> xLineTo(case_radius, %)
  |> yLineTo(0.25, %)
  |> xLineTo(throat_R + converging_radius, %)
  |> tangentialArcTo([throat_R, 0], %)
  |> close(%)
```

**Comment by nadr0** [comment]
tl;dr
@mlfarrell The sketch plane and grid plane misaligned is not a bug. The wrong starting position was picked when doing the sketch.
There appears to be a revolve engine bug and it doesn't like the shape that was provided.

Can I close this issue and we create a new one for the revolve or should we reuse this issue? 

-----

The sketch plane and grind plane misaligned is not a bug. 

The issue is that the starting point that was picked did not account for the height that was drawn above it. 
![Image](https://github.com/user-attachments/assets/d5317614-033d-4cea-8a49-047f9f9745e9)

Here is a hack I did to shift the model down by `0.25`. This is hard coded for 0.25 so it won't work with other values. 
Another way to fix this is by choosing at point on the top of the shape and drawing that content first instead of picking a point further down the shape as the starting point. 
```
// Hard coded fix to shift 0.25 down.
const throat_D = 2
const throat_R = throat_D / 2
const throat_flat = 0.1
const diverging_half_angle = 15
const diverging_radius = 0.25
const epsilon = 6
const exit_D = epsilon ^ 0.5 * throat_D
const exit_R = exit_D / 2
const cone_thickness = 0.25
const body_diam = throat_D + 1
const body_R = body_diam / 2
const converging_radius = 0.25
const case_diam = 4
const case_radius = case_diam / 2

const shift = 0.25
const sketch001 = startSketchOn("XZ")
  |> startProfileAt([throat_R, -shift], %)
  |> yLine(-throat_flat, %)
  |> arc({
       angleStart: 0,
       angleEnd: diverging_half_angle,
       radius: -diverging_radius
     }, %)
  |> angledLineToX({
       angle: diverging_half_angle - 90,
       to: exit_R
     }, %)
  |> xLine(cone_thickness, %)
  |> angledLineToX({
       angle: diverging_half_angle + 90,
       to: body_R
     }, %)
  |> yLineTo(-.25 + -shift, %)
  |> xLineTo(case_radius, %)
  |> yLineTo(0, %)
  |> xLineTo(throat_R + converging_radius, %)
  |> tangentialArcTo([throat_R, -shift], %)
  |> close(%)
```

There appears to be an engine bug if you look at the source code I provided in the previous comment. 

This will produce an engine error
```
{"kind":"engine","sourceRanges":[[1002,1034]],"msg":"Modeling command failed: [ApiError { error_code: InternalEngine, message: \"Solid3D revolve failed:  sketch profile must lie entirely on one side of the revolution axis\" }]"}
```

**Comment by mlfarrell** [comment]
oh interesting.  if there's a revolve issue for sure, then yea, create that engine-side.

**Comment by nadr0** [comment]
Thanks, I created an engine side issue for the revolve case. I will be closing this modeling app issue since there is no bug.
